### PR TITLE
Refactor/pluign garfish

### PR DESCRIPTION
**Pull request by zhoushaw**
# PR Details

Refactor Garfish Plugin

## Description

* change @modern-js/plugin-micro-frontend => @modern-js/plugin-garfish
* remove disableCustomerRouter logic
* adding unit tests

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ x] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/cli/plugin-bff/.runtime-exports/index.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export * from '/Users/bytedance/github/modern.js/packages/cli/plugin-runtime'
```
Comment (leeight) [inline packages/cli/plugin-bff/.runtime-exports/index.js:1]: `.runtime-exports` 不应该被提交进来

File: packages/cli/plugin-garfish/.vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "cSpell.words": [
```
Comment (leeight) [inline packages/cli/plugin-garfish/.vscode/settings.json:2]: 这个文件也不应该提交上来

File: packages/cli/plugin-garfish/src/runtime/utils/constant.ts [hunk 3]
```diff
@@ -2,6 +2,8 @@ import { ModuleInfo } from '../typings';
 
 export const SUBMODULE_APP_COMPONENT_KEY = 'SubModuleComponent';
 
+export const JUPITER_SUBMODULE_APP_COMPONENT_KEY = 'jupiter_submodule_app_key';
```
Comment (caohuilin) [inline packages/cli/plugin-garfish/src/runtime/utils/constant.ts:5]: jupiter 是不是改成 modern 比较好

File: packages/cli/plugin-storybook/.runtime-exports/index.js [hunk 4]
```diff
@@ -0,0 +1 @@
+export * from '/Users/bytedance/github/modern.js/packages/cli/plugin-runtime'
```
Comment (caohuilin) [inline packages/cli/plugin-storybook/.runtime-exports/index.js:1]: 这个也要删掉

**Review by caohuilin — COMMENTED** [review]
在根目录执行执行 pnpm change，加一下 changeset

File: packages/cli/plugin-garfish/src/runtime/utils/constant.ts (hunk 3 above)
Comment (zhoushaw) [reply packages/cli/plugin-garfish/src/runtime/utils/constant.ts:5]: 这个主要是做兼容内部的旧的获取方式，否则用户可能没法增量升级

File: .changeset/nasty-oranges-eat.md [hunk 5]
```diff
@@ -20,7 +20,7 @@
 "@modern-js/plugin-lambda-fc": minor
 "@modern-js/plugin-lambda-scf": minor
 "@modern-js/plugin-less": minor
-"@modern-js/plugin-micro-frontend": minor
+"@modern-js/plugin-garfish": minor
```
Comment (caohuilin) [inline .changeset/nasty-oranges-eat.md:23]: 这个文件最新分支已经没有了，changeset 是要根据改动重新生成的

File: packages/cli/plugin-garfish/src/runtime/utils/constant.ts (hunk 3 above)
Comment (zhoushaw) [reply packages/cli/plugin-garfish/src/runtime/utils/constant.ts:5]: 我在内部引用得时候默认增加一个配置吧，改成动态的

**Pull request closed without merging**